js/client.js

```diff
@@ -165,6 +166,9 @@
 					self.nameRegExp = new RegExp('(?:\\b|(?!\\w))' + escaped + '(?:\\b|\\B(?!\\w))', 'i');
 				}
 			});
+			this.on('change:settings', function () {
+				Dex.prefs('usersettings', self.get('settings'));
```

Comment:
I also don't like `usersettings` as the pref key here. All of `Storage.prefs` are user settings. These are just the settings that need to be sent to the server.

Reply:
I know I know bikeshedding, but client localStorage is literally the most permanent support burden we have. Even old server protocols we tend to stop supporting after a few years.

Reply:
`serversettings` possibly. It also might be an idea to just have them at the top level with everything else.

Reply:
I went with `serversettings`. I'd be OK with pulling them out (I assume not just at the `localStorage layer`, but also in `User`?), but this results in the simplest code, so I'd only do it if you wanted it?

> I know I know bikeshedding, but client localStorage is literally the most permanent support burden we have.

Its still not set in stone? We could write `blockchallenges` and `blockpms` settings in the future, and add migration logic which, if it detects a `serversettings` setting will perform the translation and delete the old storage. At a certain point (after a sunset period) you would want to remove the migration logic though, and users who hadn't used the app during the entire migration period would have their settings lost. Which in these edge cases, seems like an acceptable risk (oh no, they'll need to toggle the checkboxes again!)

That being said, I totally agree that protocol and naming is important. localStorage is effectively a database on the user's computer, so it takes schema migrations which are already painful and makes them 10x worse.

Reply:
I think I actually prefer `serversettings` to individually setting them. I think because we get them all from the same place (`|updateuser|`) editing them together makes sense.

That being said, 6fb7989 pulls them out of localStorage if you'd like (I left them together as `settings` in `User` because I think its nice that the'yre namespaced there instead of cluttering the top level properties) . You can revert 4b737f2 and merge that instead if you want them to be stored separately in users' browsers (though as mentioned above, whatever you decide isn't exactly set in stone).